substrate/client/network/sync/src/blocks.rs

```diff
@@ -123,20 +123,36 @@ impl<B: BlockT> BlockCollection<B> {
 		let first_different = common + <NumberFor<B>>::one();
 		let count = (count as u32).into();
 		let (mut range, downloading) = {
+			// Iterate through the ranges in `self.blocks` looking for a range to download
 			let mut downloading_iter = self.blocks.iter().peekable();
 			let mut prev: Option<(&NumberFor<B>, &BlockRangeState<B>)> = None;
 			loop {
 				let next = downloading_iter.next();
 				break match (prev, next) {
+					// If we are already downloading this range, request it from `max_parallel`
+					// peers (`max_parallel = 5` by default).
```

Comment:
I don't understand why we need to request same blocks from multiple peers simultaneously in the first place. We'll import only blocks sent by the first responding peer, other responses will be effectively discarded.

Reply:
[The comment](https://github.com/paritytech/polkadot-sdk/blob/0aeab381380107a6462fea240181f6e2a78f21b8/substrate/client/cli/src/params/network_params.rs#L112-L117) for the CLI flag explains. Slow peers could slow down syncing since each request can take up to 20 seconds. Could be worth checking how much slower sync from genesis to some block X would be if the value was set to 1. Would need a reproducible environment though.